src/components/modals/insertImageModal/partials/UploadImageForm.tsx

```diff
@@ -0,0 +1,89 @@
+import { Button, Center, Group, Image, Stack, Text } from "@mantine/core";
+import { Dropzone, FileWithPath } from "@mantine/dropzone";
+import { useMemo, useState } from "react";
+import { convertBase64 } from "../../../../lib/helpers/convertBase64";
+import { TInsertImageModalProps } from "../InsertImageModal";
+import Compressor from "compressorjs";
+
+type UploadImageFormProps = Pick<
+  TInsertImageModalProps,
+  "handleSubmit" | "handleCancel"
+>;
+
+export const UploadImageForm = ({
+  handleCancel,
+  handleSubmit,
+}: UploadImageFormProps) => {
+  const [uploadedFile, setUploadedFile] = useState<FileWithPath>();
+
+  const imagePreview = useMemo(() => {
+    if (uploadedFile) {
+      const imageUrl = URL.createObjectURL(uploadedFile);
+      return (
+        <Center mb={"lg"}>
+          <Image
+            src={imageUrl}
+            h={200}
+            w="auto"
+            maw={"100%"}
+            fit="contain"
+            onLoad={() => URL.revokeObjectURL(imageUrl)}
+          />
+        </Center>
+      );
+    }
+
+    return null;
+  }, [uploadedFile]);
+
+  const handleFormSubmit = () => {
+    if (!handleSubmit || !uploadedFile) {
+      return;
+    }
+
+    new Compressor(uploadedFile, {
+      quality: 0.6,
+      success: async (result) => {
+        const base64 = (await convertBase64(result)) as string;
+        handleSubmit(base64);
+      },
+      error(err) {
+        console.log(err.message);
+      },
+    });
+  };
+
+  return (
+    <Stack>
+      <Dropzone
+        onDrop={(files) => {
+          setUploadedFile(files[0]);
+        }}
+        maxSize={2 * 1024 ** 2}
+        accept={{
+          "image/*": [], // All images
+        }}
```

Comment:
We could move this as `IMAGE_UPLOAD_MAX_SIZE` and `IMAGE_UPLOAD_MIME_TYPES` to a config file like `src/lib/constants/generic.ts`